src/controllers/article.controller.js

```diff
@@ -124,6 +85,154 @@ class ArticleController {
       `article with id = ${articleId} has been successfully removed from your bookmarks`
     );
   }
+
+  /**
+   * @static
+   * @param {object} req - - express request parameter
+   * @param {object} res - - express response parameter
+   * @returns {object} - - article object
+   * @memberof ArticleController
+   */
+  static async createArticle(req, res) {
+    const { id } = req.currentUser;
+    const { title, description, body, image } = req.body;
+    try {
+      const article = await BaseRepository.create(db.Article, {
+        title,
+        description,
+        body,
+        image,
+        slug: slug(
+          `${title}-${crypto.randomBytes(12).toString('base64')}`
+        ).toLowerCase(),
+        authorId: id
+      });
+      return responseGenerator.sendSuccess(res, 201, article);
+    } catch (error) {
+      return responseGenerator.sendError(res, 400, error.message);
+    }
+  }
+
+  /**
+   * Get All Articles
+   * @async
+   * @param {object} req - Request object
+   * @param {object} res - Response object
+   * @return {json} Returns json object
+   * @static
+   */
+  static async fetchAllArticles(req, res) {
+    try {
+      const { page } = req.query;
+      const paginate = new Pagination(page, req.query.limit);
+      const { limit, offset } = paginate.getQueryMetadata();
+      const { count, rows: articles } = await BaseRepository.findAndCountAll(
+        db.Article,
+        {
+          limit,
+          offset
+        }
+      );
+
+      return articles.length > 0
+        ? responseGenerator.sendSuccess(
+            res,
+            200,
+            articles,
+            null,
+            paginate.getPageMetadata(count, '/api/v1/articles')
+          )
+        : responseGenerator.sendSuccess(res, 404, null, 'Articles Not Found');
+    } catch (error) {
+      return responseGenerator.sendError(res, 500, error.message);
+    }
+  }
+
+  /**
+   *
+   *
+   * @static
+   * @param {*} req - - express request parameter
+   * @param {*} res - - express response parameter
+   * @returns {object} - - article object
+   * @memberof ArticleController
+   */
+  static async fetchSpecificArticle(req, res) {
+    const { articleId } = req.params;
+    try {
+      const retrievedArticle = await BaseRepository.findOne(
+        db.Article,
+        articleId
+      );
+      return retrievedArticle
+        ? responseGenerator.sendSuccess(res, 200, retrievedArticle)
```

Comment:
Nice implementation here but could you use the middleware on the link I specified above?

Reply:
Thank you for this feedback.
I have checked the `article middleware` and it does not fit into the requirements of the task implemented here.